src/Features/Core/Portable/GenerateMember/GenerateConstructor/AbstractGenerateConstructorService.State.cs

```diff
@@ -98,18 +98,68 @@ private State()
 
             private bool ClashesWithExistingConstructor(TService service, SemanticDocument document, CancellationToken cancellationToken)
             {
-                var parameters = this.ParameterTypes.Zip(this.ParameterRefKinds, (t, r) => CodeGenerationSymbolFactory.CreateParameterSymbol(
-                    attributes: null,
-                    refKind: r,
-                    isParams: false,
-                    type: t,
-                    name: string.Empty)).ToList();
+                if (this.ParameterTypes == null ||
+                    this.ParameterRefKinds == null)
+                {
+                    return false;
+                }
+
+                int shortestLength = this.ParameterTypes.Count < this.ParameterRefKinds.Count ?
+                    this.ParameterTypes.Count :
+                    this.ParameterRefKinds.Count;
+
+                if (!this.TypeToGenerateIn.InstanceConstructors.Any(c => c.Parameters.Length == shortestLength))
+                {
+                    return false;
+                }
 
                 var destinationProvider = document.Project.Solution.Workspace.Services.GetLanguageServices(this.TypeToGenerateIn.Language);
                 var syntaxFacts = destinationProvider.GetService<ISyntaxFactsService>();
+                for (int i = 0; i < shortestLength; i++)
+                {
+                    var compareParameterName = false;
+                    var type = this.ParameterTypes[i];
+                    var refKind = this.ParameterRefKinds[i];
+                    string parameterName = GetArgument(service, i);
+                    if (!string.IsNullOrEmpty(parameterName))
+                    {
+                        compareParameterName = true;
+                    }
+                    var parameterSymbol = CodeGenerationSymbolFactory.CreateParameterSymbol(
+                        attributes: null,
+                        refKind: refKind,
+                        isParams: false,
+                        type: type,
+                        name: parameterName);
+                    var result = this.TypeToGenerateIn.InstanceConstructors.Any(c =>
```

Comment:
I'm confused by this 'Any' especially in the context of the above 'for' loop.  

Don't we want to just take the list of proposed parameters, and see if there is any constructor that matches all those proposed parmaeters?